src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker/BakingProduct.java

```diff
@@ -0,0 +1,198 @@
+    public boolean nextState()
+    {
+        if(ProductState.values().length <= state.ordinal())
+        {
+            return false;
+        }
+        state = ProductState.values()[state.ordinal()+1];
+        return true;
+    }
+
+    /**
+     * Set the State of the BakingProduct.
+     * @param state the state to set it to.
+     */
+    private void setState(final ProductState state)
+    {
+        this.state = state;
+    }
+
+    /**
+     * Getter for the end product of the bakery.
+     * @return the itemStack of the product.
+     */
+    public ItemStack getEndProduct()
+    {
+        return endProduct;
+    }
+
+    /**
+     * Get the Baking progress of the product in the oven.
+     * @return an integer between 0-10.
+     */
+    public int getBakingProgress()
+    {
+        return bakingProgress;
+    }
+
+    /**
+     * Increase the baking progress.
+     * If reached finish line, this will make the product reach the next state.
+     */
+    public void increaseBakingProgress()
+    {
+        this.bakingProgress++;
+
+        if(bakingProgress >= FINISHED_BAKING_PROGRESS)
+        {
+            this.nextState();
+        }
+    }
+
+    /**
+     * Getter for the recipe id of the product.
+     * @return the id.
+     */
+    public int getRecipeId()
+    {
+        return recipeId;
+    }
+
+    /**
+     * Create the product from NBT.
+     * @param productCompound the compound to use.
+     * @return the restored BakingProduct.
+     */
+    public static BakingProduct createFromNBT(final NBTTagCompound productCompound)
+    {
+        if(productCompound.hasKey(TAG_STATE))
+        {
+            final ProductState state = ProductState.values()[productCompound.getInteger(TAG_STATE)];
+            final int recipeId = productCompound.getInteger(TAG_RECIPE_ID);
+            final BakingProduct bakingProduct = new BakingProduct(ItemStack.loadItemStackFromNBT(productCompound), recipeId);
+            bakingProduct.setState(state);
+            return bakingProduct;
+        }
+        return null;
+    }
+
+    /**
+     * Write the BakingProduct to NBT.
+     * @param productCompound the compound to write it to.
+     */
+    public void writeToNBT(final NBTTagCompound productCompound)
+    {
+        productCompound.setInteger(TAG_STATE, state.ordinal());
+        productCompound.setInteger(TAG_RECIPE_ID, recipeId);
+        endProduct.writeToNBT(productCompound);
+    }
+
+    @Override
+    public boolean equals(final Object o)
+    {
+        if (this == o)
+        {
+            return true;
+        }
+        if (!(o instanceof BakingProduct))
+        {
+            return false;
+        }
+
+        final BakingProduct bakingProduct = (BakingProduct) o;
+
+        if (recipeId != bakingProduct.recipeId)
+        {
+            return false;
+        }
+        if (state != bakingProduct.state)
+        {
+            return false;
+        }
+        if (!endProduct.equals(bakingProduct.endProduct))
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1126)